java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/ActionPlanActivity.java

```diff
@@ -0,0 +1,133 @@
+import android.widget.ListView;
+import android.widget.Spinner;
+import android.widget.Toast;
+import android.widget.AdapterView.OnItemSelectedListener;
+
+import healthvault.client.implementation.MicrosoftHealthVaultRestApiImpl;
+import healthvault.client.models.ActionPlan;
+import healthvault.client.models.ActionPlanInstance;
+import healthvault.client.models.ActionPlansResponseActionPlanInstance;
+
+import healthvault.client.models.Objective;
+import okhttp3.ResponseBody;
+import retrofit2.Response;
+import rx.Subscriber;
+import rx.Subscription;
+import rx.schedulers.Schedulers;
+
+public class ActionPlanActivity  extends Activity {
+	private HealthVaultApp mService;
+	private HealthVaultClient mClient;
+	private Record mCurrentRecord;
+	private ArrayAdapter<String> mAdapter;
+	private ListView mPlanList;
+	private static ActionPlansResponseActionPlanInstance mActionPlanInstance;
+
+	@Override
+	public void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_actionplan);
+		mService = HealthVaultApp.getInstance();
+		mClient = new HealthVaultClient();
+
+		if (mService.isAppConnected()) {
+			mCurrentRecord = HealthVaultApp.getInstance().getCurrentRecord();
+			getActionPlan();
+		}
+	}
+
+	@Override
+	protected void onStart() {
+		super.onStart();
+		mClient.start();
+	}
+
+	@Override
+	protected void onStop() {
+		mClient.stop();
+		super.onStop();
+	}
+
+	@SuppressWarnings("unchecked")
+	private void getActionPlan() {
+		mService.getSettings().setRestUrl("https://data.ppe.microsofthealth.net/");
+		HealthVaultSettings settings = mService.getSettings();
+		Connection connection = mService.getConnection();
+
+		final MicrosoftHealthVaultRestApiImpl restClient = new HealthVaultRestClient(settings, connection, mCurrentRecord).getClient();
+
+		restClient.getActionPlansAsync()
+				.subscribeOn(Schedulers.io())
+				.observeOn(Schedulers.io())
+				.subscribe(new Subscriber<Object>() {
+					@Override
+					public final void onCompleted() {
+					}
+
+					@Override
+					public final void onError(final Throwable e) {
+						Log.e("error", e.getMessage());
+					}
+
+					@Override
+					public final void onNext(final Object response) {
+						mActionPlanInstance = (ActionPlansResponseActionPlanInstance) response;
+					}
+				});
+		renderActionPlans();
+	}
+
+	private void renderActionPlans() {
+		List<String> actionplans = new ArrayList<String>();
+		if (mActionPlanInstance != null) {
+			int size = mActionPlanInstance.plans().size();
+			for (int index = 0; index < size; ++index) {
+				ActionPlanInstance plan = mActionPlanInstance.plans().get(index);
+				actionplans.add("Plan: " + plan.name().toString() +
+						"     Category: " + plan.category().toString());
+			}
+			mPlanList = (ListView) findViewById(R.id.actionplan_list);
+			mAdapter = new ArrayAdapter<String>(ActionPlanActivity.this, android.R.layout.simple_list_item_1, actionplans);
+			mPlanList.setAdapter(mAdapter);
```

Comment:
We should only be setting the adapter once and just updated the contents of that adapter when they change then notifyDatasetChanged(). I think you have done this before in other reviews so do the same here.